### PR TITLE
Add prompt to `wrangler pages download config`

### PR DESCRIPTION
**Pull request by penalosa**
## What this PR solves / how to test

Adds an interactive prompt to `wrangler pages download config` if downloading the config would overwrite an existing `wrangler.toml` file.

Fixes [DEVX-1243](https://jira.cfdata.org/browse/DEVX-1243)

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [x] TODO (before merge)
  - [ ] Included
  - [ ] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): <https://github.com/cloudflare/cloudflare-docs/pull/>...
  - [x] Not necessary because: Not yet documented

### CODE REVIEW

File: .changeset/wise-grapes-prove.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"wrangler": minor
+---
+
+feat: Add interactive prompt to `wrangler pages download config` if an exsting `wrangler.toml` file exists
```
Comment (petebacondarwin) [inline .changeset/wise-grapes-prove.md:5]:
```suggestion
feat: Add interactive prompt to `wrangler pages download config` if an existing `wrangler.toml` file exists
```

File: packages/wrangler/src/__tests__/pages/pages-download-config.test.ts [hunk 2]
```diff
@@ -447,4 +456,329 @@ describe("pages download config", () => {
...
+					  [[env.production.analytics_engine_datasets]]
+					  binding = \\"AE\\"
+					  dataset = \\"data\\"
+
+					  [env.production.ai]
+					  binding = \\"AI\\"
+					"
+			`);
+		});
+		it("should not overwrite existing file w/o --force (non-interactive)", async () => {
```
Comment (petebacondarwin) [inline packages/wrangler/src/__tests__/pages/pages-download-config.test.ts:602]: Should we check what messages are logged in this case?

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
Is `false` the correct default option here for non-interactive?
I feel like it should actually fail... because otherwise if someone is expecting their CI to download and keep the wrangler.toml updated before deploying, and haven't specified `--force` it will just deploy the old wrangler.toml instead.

**Review by petebacondarwin — APPROVED** [review]

File: packages/wrangler/src/pages/download-config.ts [hunk 3]
```diff
@@ -225,7 +225,7 @@ export const Handler = async ({ projectName, force }: DownloadConfigArgs) => {
 			{ fallbackValue: false }
 		);
 		if (!overwrite) {
-			return;
+			throw new FatalError("Not overwriting existing `wrangler.toml` file");
```
[the remark this answers is not in the document]
Comment (admah) [reply packages/wrangler/src/pages/download-config.ts:228]: I think it's ok for now. We can revisit if needed.

**Pull request merged**